firstvoices/backend/serializers/media_serializers.py

```diff
@@ -89,20 +90,44 @@ class Meta(MediaWithThumbnailsSerializer.Meta):
 
 
 class VideoSerializer(MediaWithThumbnailsSerializer):
-    """Serializer for Audio objects. Supports video objects shared between different sites."""
+    """Serializer for Video objects. Supports video objects shared between different sites."""
 
     original = MediaVideoFileSerializer()
 
     class Meta(MediaWithThumbnailsSerializer.Meta):
         model = media.Video
 
 
+class WriteableRelatedAudioSerializer(serializers.PrimaryKeyRelatedField):
+    def to_representation(self, value):
+        return AudioSerializer(context=self.context).to_representation(value)
+
+
+class WriteableRelatedVideoSerializer(serializers.PrimaryKeyRelatedField):
+    def to_representation(self, value):
+        return VideoSerializer(context=self.context).to_representation(value)
+
+
+class WriteableRelatedImageSerializer(serializers.PrimaryKeyRelatedField):
+    def use_pk_only_optimization(self):
+        return False
+
+    def to_representation(self, value):
+        return ImageSerializer(context=self.context).to_representation(value)
+
```

Comment:
🙌 These are nice!

Why does the image one skip the pk-only optimization?

Reply:
They probably all should, to be safe, and I've updated the code accordingly. If that optimization is enabled (as it is by default), the serializer receives a "stub" object that is missing properties the serializer might reference.